main.go

```diff
@@ -83,6 +83,20 @@ func startTraceHandling(ctx context.Context, rep otelreporter.TraceReporter,
 	return err
 }
 
+func appendEndpoint(symbolEndpoints *[]reporter.SymbolEndpoint, site, apiKey, appKey string) {
+	exists := false
+	// Ensure this exact endpoint has not already been added.
+	for _, endpoint := range *symbolEndpoints {
+		if site == endpoint.Site && apiKey == endpoint.APIKey && appKey == endpoint.AppKey {
+			exists = true
+			break
+		}
+	}
+	if !exists {
+		*symbolEndpoints = append(*symbolEndpoints, reporter.SymbolEndpoint{Site: site, APIKey: apiKey, AppKey: appKey})
+	}
+}
+
```

Comment:
Very slight nit, feel free to ignore: since `append` in Go is implemented as `append([]T, ...T) []T` and not `append(*[]T, ...T)` I think it would be slightly more idiomatic to implement as such:

```suggestion
func appendEndpoint(symbolEndpoints []reporter.SymbolEndpoint, site, apiKey, appKey string) symbolEndpoints []reporter.SymbolEndpoint {
	// Ensure this exact endpoint has not already been added.
	for _, endpoint := range symbolEndpoints {
		if site == endpoint.Site && apiKey == endpoint.APIKey && appKey == endpoint.AppKey {
			return symbolEndpoints
		}
	}
	return append(symbolEndpoints, reporter.SymbolEndpoint{Site: site, APIKey: apiKey, AppKey: appKey})
}
```

Reply:
I was not sure whether we wanted to pass the list by value, hence the pointer. I will update that asap. Thanks!